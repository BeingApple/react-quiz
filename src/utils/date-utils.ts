import moment from "moment"
import { Moment, unitOfTime } from "moment"

type Unit = {
  unit: unitOfTime.Base
  string: string
}

const units: Array<Unit> = [
  {unit: "years", string: "년"},
  {unit: "months", string: "개월"}, 
  {unit: "days", string: "일"}, 
  {unit: "hours", string: "시간"}, 
  {unit: "minutes", string: "분"},
  {unit: "seconds", string: "초"},
]

export const durationToString = (duration: moment.Duration) => {
  let string = ""

  for (const unit of units) {
    const unitTime = duration.get(unit.unit)
    if (unitTime > 0) {
      string += unitTime + unit.string + " "
    }
  }

  return string.trim();
}

export const dateToString = (date?: Moment) => {
  date = (typeof date === 'string' || date instanceof String) ? moment(date) : date

  return date?.format("yyyy-MM-DD HH:mm:ss") ?? ""
}