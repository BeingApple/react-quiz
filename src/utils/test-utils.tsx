import React, { PropsWithChildren } from 'react'
import { render } from '@testing-library/react'
import type { RenderOptions } from '@testing-library/react'
import { Provider as ReduxProvider } from 'react-redux'

import { setupStore, type AppStore, type RootState } from '@/store/store'
// As a basic setup, import your same slice reducers
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>
  store?: AppStore
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  const queryClient = new QueryClient()

  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <ReduxProvider store={store}>
        <QueryClientProvider client={queryClient}>
          {children}
        </QueryClientProvider>
      </ReduxProvider>
    )
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}