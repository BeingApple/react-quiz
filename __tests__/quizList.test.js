import '@testing-library/jest-dom'
import { fireEvent } from '@testing-library/react'
import QuizList from '@/components/domain/quiz'
import { renderWithProviders } from '../src/utils/test-utils'

describe('<QuizList />', () => {
  it('matches snapshot', () => {
    const utils = renderWithProviders(<QuizList />);
    expect(utils.container).toMatchSnapshot();
  });

  it('shows the first state', () => {
    const utils = renderWithProviders(<QuizList />);
    const startButton = utils.getByText('퀴즈 풀기')

    fireEvent.click(startButton)
  })
});