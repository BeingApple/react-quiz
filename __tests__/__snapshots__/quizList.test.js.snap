// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuizList /> matches snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-9y1egq-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    퀴즈 풀기
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
